### PR TITLE
PKG-1722 update to 20230802.0

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
dependency for protobuf & grpc
- update to latest tag - 20230802.0
- add tests if libabsl_log_initialize exist - needed by protobuf >=4.0.0
- [upstream](https://github.com/abseil/abseil-cpp/tree/20230802.0)

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+cxx_compiler:               # [win]
+  - 'vs2019'                # [win]
+c_compiler:                 # [win]
+  - 'vs2019'                # [win]
+vc:                         # [win]
+  - 16                      # [win]
+
+CONDA_BUILD_SYSROOT:       # [osx and x86]
+  - /opt/MacOSX10.13.sdk   # [osx and x86]
+macos_min_version:         # [osx and x86]
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/conda_build_config.yaml:19]: @boldorider4 if you wonder about the solution for the problem we talked about, here it is

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 source:
   url: https://github.com/abseil/{{ name }}/archive/{{ version }}.tar.gz
-  sha256: dcf71b9cba8dc0ca9940c4b316a0c796be8fab42b070bb6b7cab62b48f0e66c4
+  sha256: 59d2976af9d6ecf001a81a35749a6e551a335b949d34918cfade07737b9d93c5
   patches:
     #- linux-librt.patch  # [linux]
     - clang4_osx_fix.diff  # [osx]
-    - osx_framework_fix.diff  # [osx]
-    - remove-4221.patch  # [win]
+    # - osx_framework_fix.diff  # [osx]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: You could remove the patches from the repo if we don't use them anymore.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
+    # - remove-4221.patch  # [win]
```
Comment (ViridianMelody) [inline recipe/meta.yaml]: If the only remaining patch is for osx, we should remove `m2-patch` and set `patch` to osx only.

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

**Review by ViridianMelody — APPROVED** [review]
Looks good to me aside from one comment.

**Review by akabanovs — APPROVED** [review]
LGTM

**Pull request merged**